File: CLRS Algorithms/Console C++/Sort Algorithms/Bubble Sort.cpp

#include <iostream>

using namespace std;

void Bubble_Sort(int *A , int n);
void swap(int *a, int *b);

int main(){
	int A[100];
	int n;
	cout << "Element(s) number : ";
	cin >> n;
	for(int i=0;i<n;i++){
		cin >> A[i];
	}
	system("clear");
	cout << "Current Array: " << endl;
	for(int i=0;i<n;i++){
		cout << "[ " << A[i] << " ] ";
	}
	Bubble_Sort(A,n);
	cout << endl << "Sorted Array : " << endl;
	for(int i=0;i<n;i++){
		cout << "[ " <<A[i] << " ] ";
	}
	cout <<endl;
	return 0;
}

void swap(int *a, int *b) {
	        int *temp;
	        temp = a;
	        a = b;
	        b = temp;
}

void Bubble_Sort(int *a, int n){
	for(int i=0;i<n;i++){
		for(int j=n-1;j>=i+1;j--){
			if(a[j]<a[j-1]){
				swap(a[j],a[j-1]);
			}
		}
	}
}

File: CLRS Algorithms/Python/Sort Algorithms/Bubble Sort.py
#   Poorya Mohammadi Nasab
#   April,2019

import os

def Bubble_Sort(Array=[],n=0):
    #begin
    for i in range(n-1):
        for j in range(n-i-1):
            # begin
            if Array[j] > Array[j+1]:
                tmp = Array[j]
                Array[j]=Array[j+1]
                Array[j+1]=tmp
            # end
    return Array
    #end
    

# main

Array = []
while True:
    # begin
    try:
        n = int(input('.:: Element(s) number : '))
    except :
        print('Error : You must enter a positive integer !!')
        continue
    else :
        if n<=0:
            print('Error : You must enter a positive integer !!')
        else :
            break
    # end

for i in range(n):
    Array.append(float(input('.:: Enter num[ {} ] : '.format(i+1))))

os.system('clear')
print(Array)
Array = Bubble_Sort(Array,n)
print(Array)
# end main 